File: C/Cap07/c07ex06.c
// c07ex06.c

#include <stdio.h>

int FATORIAL(int N) {
  int Fat = 1;
  for (int I = 1; I <= N; I++) {
    Fat *= I;
  }
  return Fat;
}

int main(void) {
  int Limite;

  printf("Qual fatorial: ");
  scanf("%d", &Limite);
  while (getchar() != '\n');

  printf("Fatorial = %d\n", FATORIAL(Limite));

  printf("\n");
  printf("Tecle <Enter> para encerrar... ");
  while (getchar() != '\n');

  return 0;
}

File: TYPESCRIPT/Cap07/c07ex06.ts
// c07ex06.ts

import * as rls from 'readline-sync';

function FATORIAL(N: number): number {
  let Fat: number = 1;
  for (let I: number = 1; I <= N; I++) {
    Fat *= I;
  }
  return Fat;
}

let Limite: number;

Limite = rls.questionInt("Qual fatorial: ");

console.log(`Fatorial = ${FATORIAL(Limite)}`);

console.log("");
process.stdout.write("Tecle <Enter> para encerrar... ");
rls.question();
